collect_app/src/main/java/org/odk/collect/android/widgets/IntegerWidget.java

```diff
@@ -74,7 +74,7 @@ public IntegerWidget(Context context, FormEntryPrompt prompt, boolean readOnlyOv
         Integer i = getIntegerAnswerValue();
 
         if (i != null) {
-            mAnswer.setText(i.toString());
+            mAnswer.setText(String.format("%d", i.toString()));
```

Comment:
I think you need to remove the call on toString() for it to be an improvement. I'm surprised that this works -- %d doesn't require the param to be an int? I think %f is for a double.